File: solutions/179.Largest_Number/AC_sort_nlogn.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_sort_nlogn.py
# Create Date: 2015-02-26 10:33:39
# Usage:       AC_sort_nlogn.py 
# Descripton:  


class Solution:
    # @param num, a list of integers
    # @return a string
    def largestNumber(self, num):
        comp = lambda a, b: -1 if a + b > b + a else 1 if a + b < b + a else 0
        return str(int("".join(sorted(map(str, num), comp))))

# debug
s = Solution()
print s.largestNumber([0, 0, 5])

File: solutions/217.Contains_Duplicate/AC_sort_nlogn.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_sort_nlogn.cpp
*  Create Date: 2015-07-13 21:41:31
*  Descripton:   
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        return unique(nums.begin(), nums.end()) != nums.end();
    }
};

int main() {

    return 0;
}

